### PR TITLE
add unnecessary_library_name to recommended

### PR DESCRIPTION
**Pull request by devoncarew**
- add `unnecessary_library_name` to the recommended lint set
- close https://github.com/dart-lang/lints/issues/181
- bump the min sdk to the first stable containing this lint

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 5.0.0-wip
+
+- `recommended`:
+  - added `unnecessary_library_name` (https://github.com/dart-lang/lints/issues/181)
```
Comment (parlough) [inline CHANGELOG.md:4]: Consider linking the lint name to https://dart.dev/lints/unnecessary_library_name so readers of the changelog can learn more :)

**Review by parlough — APPROVED** [review]
Nice! I imagine this will be a big simplicity and education win for lots of projects :)

**Review by kevmoo — APPROVED** [review]
Sweet!

**Review by pq — APPROVED** [review]
🚀

**Pull request merged**